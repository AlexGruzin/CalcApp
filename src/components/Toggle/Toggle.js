import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Toggle.styles.less';

const Toggle = ({ className, label, name, value, checked, ...innerProps }) => (
  <div className="custom-toggle-wrapper">
    {label && (
      <label className="custom-toggle-label" htmlFor={name}>
        {label}
      </label>
    )}
    <input
      className={classNames('custom-toggle', className)}
      type="checkbox"
      checked={!!(checked || value)}
      name={name}
      id={name}
      {...innerProps}
    />
  </div>
);

Toggle.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.bool,
  checked: PropTypes.bool,
};

Toggle.defaultProps = {
  className: '',
  label: '',
  name: '',
  value: false,
  checked: false,
};

export default Toggle;
