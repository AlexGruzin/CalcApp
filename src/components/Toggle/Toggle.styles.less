.custom-toggle[type="checkbox"]{
  position: relative;
  height: 15px;
  width: 30px;
  min-width: 30px;
  -webkit-appearance: none;
  background: var(--grey300);
  outline: none;
  border-radius: 15px;
  transition: .2s;
  border: 1px solid var(--grey);
  cursor: pointer;
}

.custom-toggle-wrapper {
  display: flex;
}

.custom-toggle-label {
  font-family: "MontserratBold", sans-serif;
  letter-spacing: 0.25px;
  font-size: var(--fontSize-tiny);
}

.custom-toggle:checked[type="checkbox"]{
  background: var(--purple);
}

.custom-toggle[type="checkbox"]:before{
  content: '';
  position: absolute;
  height: 11px;
  width: 11px;
  border-radius: 10px;
  top: 1px;
  left: 1px;
  background: #fff;
  transform: scale(1.1);
  transition: .5s;
}

.custom-toggle:checked[type="checkbox"]:before{
  left: 16px;
}