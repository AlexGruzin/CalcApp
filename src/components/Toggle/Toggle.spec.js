import React from 'react';
import { waitFor, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Toggle from './Toggle';

describe('Toggle tests', () => {
  it('Check snapshots of Toggle are equal.', () => {
    const { asFragment } = render(<Toggle name="name" label="custom" value onChange={() => undefined} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Toggle do change value with onChange.', async () => {
    const mockOnChange = jest.fn();
    const { getByLabelText } = render(<Toggle name="name" label="custom" onChange={mockOnChange} />);

    const input = getByLabelText('custom');
    userEvent.click(input);

    await waitFor(() => {
      expect(mockOnChange).toBeCalledTimes(1);
    });
  });
});
