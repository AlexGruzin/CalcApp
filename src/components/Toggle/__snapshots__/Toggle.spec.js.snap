// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle tests Check snapshots of Toggle are equal. 1`] = `
<DocumentFragment>
  <div
    class="custom-toggle-wrapper"
  >
    <label
      class="custom-toggle-label"
      for="name"
    >
      custom
    </label>
    <input
      checked=""
      class="custom-toggle"
      id="name"
      name="name"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
